packages/fxa-auth-server/lib/routes/recovery-phone.ts

```diff
@@ -139,6 +139,30 @@ class RecoveryPhoneHandler {
 
     return {};
   }
+
+  async exists(request: AuthRequest) {
+    const { uid, emailVerified, mustVerify, tokenVerified } = request.auth
+      .credentials as any;
+
+    // Short circuit if the account / session still needs verification.
+    // Note this is typically due to totp being required, but there are
+    // other states that could also result in an unverified session, such
+    // as a forced password change.
+    if (emailVerified || (mustVerify && !tokenVerified)) {
+      return {};
+    }
+
+    try {
+      return await this.recoveryPhoneService.hasConfirmed(uid);
+    } catch (error) {
+      throw AppError.backendServiceFailure(
```

Comment:
hmmm we've been using this for failures from service dependencies external to auth-server (Stripe, Google oauth, other Mozilla/FxA [micro]services, etc.).  Any reason for not throwing an ol' boring error like other auth-server route handlers.

Reply:
I was viewing the `recoveryPhoneService` as a back end service. In a lot of cases the `recoveryPhoneService` is wrapping calls to twilio, which is similar to a call to stripe, google, etc... In this particular case we are just making a DB call, but it felt more consistent to handle any issue with a call to the `recoveryPhoneService` as a back end service error. At least this is how I was thinking about it. Throwing a more general error also seems fine...

Reply:
@chenba I'm going to go ahead and land this, but happy to use a different error if necessary. Just let me know what you think, and I can do a quick follow up.

Reply:
> In a lot of cases the recoveryPhoneService is wrapping calls to twilio, which is similar to a call to stripe, google, etc...

ah, that make sense.  Thanks.